data-raw/simulate-data.R

```diff
@@ -0,0 +1,240 @@
+  }
+
+  if (date >= "1994-01-01") {
+    create_fake_icd10(n)
+  } else {
+    create_fake_icd8(n)
+  }
+}
+
+create_fake_icd8 <- function(n) {
+  here("data-raw/icd8-codes.txt") |>
+    read_lines() |>
+    str_trim() |>
+    as_tibble() |>
+    separate_wider_delim(
+      value,
+      delim = regex("   +"),
+      names = c("icd8", "description", "unknown"),
+      too_many = "merge"
+    ) |>
+    mutate(across(everything(), str_trim)) |>
+    mutate(
+      description = str_remove(description, "\\d+"),
+      icd8 = str_remove(icd8, "dia")
+    ) |>
+    pull(icd8) |>
+    sample(size = n, replace = TRUE)
+}
+
+create_fake_icd10 <- function(n) {
+  # from: https://medinfo.dk/sks/brows.php?s_nod=6308
+  here("data-raw/icd10-codes.csv") |>
+    read_csv2(show_col_types = FALSE) |>
+    pull(icd10) |>
+    sample(size = n, replace = TRUE)
+}
+
+create_fake_atc <- function(n) {
+  codeCollection::ATCKoodit |>
+    tibble::as_tibble() |>
+    dplyr::filter(stringr::str_length(Koodi) == 7) |>
+    dplyr::pull(Koodi) |>
+    sample(n, replace = TRUE)
+}
+
+create_fake_date <- function(n, from = "1977-01-01", to = lubridate::today()) {
+  seq(as_date(from), as_date(to), by = "day") |>
+    sample(n, replace = TRUE)
+}
+
+create_padded_integer <- function(n, length) {
+  purrr::map(1:length, \(ignore) sample(0:9, n, replace = TRUE)) |>
+    purrr::reduce(\(integer1, integer2) paste(integer1, integer2, sep = "")) |>
+    pad_integers(width = length)
+}
+
+create_fake_npu <- function(n) {
+  stringr::str_c(
+    "NPU",
+    create_padded_integer(n, 8)
+  )
+}
+
+create_fake_hovedspeciale_ans <- function(n) {
+  "https://www.dst.dk/da/Statistik/dokumentation/Times/forebyggelsesregistret/spec" |>
+    read_html() |>
+    html_element("table") |>
+    html_table() |>
+    pull(Tekst) |>
+    sample(n, replace = TRUE)
+}
+
+create_fake_drug_name <- function(atc) {
+  codeCollection::ATCKoodit |>
+    tibble::as_tibble() |>
+    dplyr::filter(Koodi %in% atc) |>
+    dplyr::pull(en) |>
+    sample(length(atc), replace = TRUE)
+}
+
+to_wwyy <- function(x) {
+  format(lubridate::as_date(x), format = "%W%y")
+}
+
+to_yyyymmdd <- function(x) {
+  format(lubridate::as_date(x), format = "%Y%m%d")
+}
+
+# Insert extra values ------------------------------------------------------
+
+insertion_rate <- function(proportion) {
+  runif(1) < proportion
+}
+
+insert_specific_atc <- function(data, proportion = 0.3) {
+  # c(
+  #       "A10AB",
+  #       "A10AC",
+  #       "A10AD",
+  #       "A10AE",
+  #       "A10BA",
+  #       "A10BB",
+  #       "A10BD",
+  #       "A10BG",
+  #       "A10BH",
+  #       "A10BJ",
+  #       "A10BK",
+  #       "A10BX"
+  # )
+  data |>
+    mutate(
+      across(
+        matches("^atc$"),
+        \(column) if_else(
+          runif(n()) < proportion,
+          sample(c("A10BA02", "A10BJ02", "A10BJ06"), 1),
+          column
+        )
+      )
+    )
+}
+
+# Insert a few false-positive cases with purchases of metformin:
```

Comment:
In not sure what you mean by false positives in this context? (same goes for L169)

Reply:
I'm not sure either haha I took it from the code @Aastedet added previously